src/Core/Usecase/Message/ReceiveMessage.php

```diff
@@ -148,16 +150,19 @@ private function createIncomingMessage($incoming_message, $contact_id, $survey_s
 	 */
 	private function createOutgoingMessage($contact_id, $survey_state_entity, $next_form_attribute)
 	{
-		// create message that we will send to thhe user next
+        // @FIXME: Message type should be configurable per deployment,survey
+        $message_type = 'sms';
+        $data_provider = \DataProvider::getEnabledProviderForType($message_type);
```

Comment:
I the case we don't know the provider, data providers were actually supposed to choose a provider at sending time. I suspect thats broken somewhere :(